### ISSUE TITLE
circular reference on security.access.decision_manager

### ISSUE THREAD

**Issue by matthbull** [symfony/symfony#21701]
| Q                | A
| ---------------- | -----
| Bug report?      | yes (possibly)
| Feature request? | no
| BC Break report? | unsure
| RFC?             | no
| Symfony version  | 2.8.15

While creating a custom voter as per [these instructions](http://symfony.com/doc/2.8/security/voters.html) I'm getting the following error when trying to inject the decision manager service:

**with public in the service declaration: true**

> Circular reference detected for service "security.access.decision_manager", path: "security.access.decision_manager".

**with public in the service declaration: false**

> Circular reference detected for service "security.access.decision_manager", path: "cache_warmer -> twig -> security.authorization_checker -> security.access.decision_manager".

according to the docs, this should only **not** work < 2.8?  However, when I then tried injecting the container, and pull out the authorization.checker service inside the voter, and I get a similar circular reference error, even though the docs say this is the workaround.

So, I'm not sure if the docs are wrong, I've missed something, or this is a bug?

**service declaration**

```
app.form_config.voter:
        class: AppBundle\Services\voters\ConfigAccessVoter
        arguments: ["@security.access.decision_manager"]
        tags:
            - { name: security.voter }
        public: false
```

**Comment by javiereguiluz** [comment]
I think the docs are wrong. The commit that merged this: https://github.com/symfony/symfony/commit/cc8c5b51819e4a5b3ce8a905ccedb2807bc80419 says:

```
This PR was squashed before being merged into the 3.1-dev branch
```

So this is not available in 2.8.

**Comment by javiereguiluz** [comment]
Let's close this as a "doc issue". We're fixing it in https://github.com/symfony/symfony-docs/pull/7524  @matthbull thanks for helping us spot this error.
